src/surveys/components/RangeResponse.tsx

```diff
@@ -0,0 +1,144 @@
+import classnames from 'classnames';
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+import { Button, Header } from '../../components';
+import { ID, Question, RangeSettings, SurveyAnswer, Theme } from '../../types';
+import RatingIcon from './RatingIcon';
+
+interface RangeResponseProps {
+  question: Question;
+  onAnswered: (questionId: ID, answer: SurveyAnswer) => void;
+  theme?: Theme;
+}
+
+function RangeResponse({ question, onAnswered, theme }: RangeResponseProps) {
+  const [value, setValue] = useState(0);
+  const [answerId, setAnswerId] = useState<ID | null>(null);
+
+  useEffect(() => {
+    const answerQuestion = () => {
+      if (value === 0) return;
+
+      if (answerId)
+        return onAnswered(question.id, {
+          type: question.type,
+          id: answerId,
+        });
+
+      onAnswered(question.id, {
+        type: question.type,
+        value: value.toString(),
+      });
+    };
+
+    const timeoutId = setTimeout(answerQuestion, 500);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [value, answerId]);
+
+  let rangeContent = null;
+
+  if (question.options && question.options.length > 0) {
+    rangeContent = question.options
+      .sort((a, b) => a.orderNumber - b.orderNumber)
+      .map((option) => {
+        if (question.type === 'nps')
+          return (
+            <Button
+              label={option.orderNumber.toString()}
```

Comment:
@AfeezGL Using `orderNumber` here is not the correct behaviour because it's not guaranteed to be sequential. We could have 2,4,8,10,16... you get the point? So it's better to use the index.